server/entity/Player.gd

```diff
@@ -108,4 +116,19 @@ func take_damage(x):
 		health = MAX_HEALTH/2
 		mana = MAX_MANA/2
 		stamina = MAX_STAMINA/2
-		
\ No newline at end of file
+		
+		rpc("set_health", health)
+		rpc("set_mana", mana)
+		rpc("set_stamina", stamina)
+
+
+remote func restore_stats(id):
+	if id == 0:
+		health += 60
```

Comment:
lots of health lol

Reply:
I think health += 60 is problematic because it doesn't account for the upper limit, so it can potentially give you more than max health.

Reply:
It's dealt with in the next line lol
```
+		if (health > MAX_HEALTH):
+			health = MAX_HEALTH
```